src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -874,6 +877,39 @@ public void update() {
             });
         });
 
+        actions.put(Actions.OPEN_CONSOLE, (BaseAction) () -> {
+            Runtime runtime = Runtime.getRuntime();
+            int selectedIndex = this.frame.tabbedPane.getSelectedIndex();
+            BasePanel bp = (BasePanel) this.frame.tabbedPane.getComponentAt(selectedIndex);
+            String path = bp.getBibDatabaseContext().getDatabaseFile().getAbsolutePath();
+            String abs_dir = path.substring(0, path.lastIndexOf(File.separator) + 1);
+            if (OS.LINUX) {
+                Process p = runtime.exec("readlink /etc/alternatives/x-terminal-emulator");
+                BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+
+                String def_term_emul = reader.readLine();
+
+
+                if (def_term_emul != null) {
+                    def_term_emul = def_term_emul.substring(def_term_emul.lastIndexOf(File.separator) + 1);
+
+                    if (def_term_emul.contains("gnome")) {
+                        runtime.exec("gnome-terminal --working-directory=" + abs_dir);
+                    } else if (def_term_emul.contains("xfce4")) {
+                        runtime.exec("xfce4-terminal --working-directory=" + abs_dir);
+                    } else if (def_term_emul.contains("konsole")) {
+                        runtime.exec("konsole --workdir=" + abs_dir);
+                    } else {
+                        runtime.exec(def_term_emul, null, new File(abs_dir));
+                    }
+                }
+            } else if (OS.WINDOWS) {
+                Runtime.getRuntime().exec("cmd.exe /c start", null, new File(abs_dir));
```

Comment:
Why don't you use the runtime variable you created at the start of the methode? Same goes for os_x case